### PR TITLE
Replace Either with Result monad

### PR DESCRIPTION
**Pull request by citizen428**
@flash-gordon This is a PR addressing our discussion in #45. I played around with various things over the last 2 weeks, but none of them were satisfactory. In the end it seems like having separate classes (though they are mostly copy and paste) is the best way forward.

Of course this means code is duplicated, but at least `RightBiased::Right` and `RightBiased::Left` could be re-used unmodified. I do believe the code duplication is justifiable in this instance, since there are lots of small places that do in fact change (`#inspect` / `#to_s` etc.). This way people can use whichever they prefer and maybe in the long run we can settle on a "winner". Or even better, is there a way to setup a custom equalizer that makes `Either` and `Result` essentially interchangeable, i.e. `Right == Success` and `Left == Failure`?

Maybe we should also remove the `success?` and `failure?` aliases from `Either`, but I'll leave that decision up to you.

/cc @timriley

### CODE REVIEW

File: lib/dry/monads.rb [hunk 1]
```diff
@@ -39,5 +40,17 @@ def Right(value)
...
+
+    # @param value [Object] the value to be stored in the monad
+    # @return [Result::Success]
+    def Success(value)
+      Result::Success.new(value)
+    end
+
+    # @param value [Object] the value to be stored in the monad
+    # @return [Result::Failure]
+    def Failire(value)
```
Comment (scepticulous) [inline lib/dry/monads.rb]: I guess it should be `Failure` not `Failire`, shouldn't it?

Comment (citizen428) [reply lib/dry/monads.rb]: It should indeed, thanks 😃

**Comment by flash-gordon** [comment]
@citizen428 thanks for this. Looking at the code I tend to agree with you that it’s better to duplicate it rather than unify them in some ugly way. But now I started to think that’s the whole thing isn’t worth the efforts and it’s better to rename Left/Right to Failure/Success just like folks in Rust did instead of keeping two variants of practically the same code. It’s very unlikely anyone would miss them and we’ll just have helpers for backward compatibility, as in `Dry::Monads.Right()` will return an instance of `Success` so nothing’s gonna break. What do you think?

**Comment by timriley** [comment]
@flash-gordon I think that'd be great 👍

**Comment by citizen428** [comment]
@flash-gordon That's fine by me. I can update this PR to remove `Either` and change the "constructors" as you suggested. Do you want me to do that?

**Comment by flash-gordon** [comment]
@citizen428 that’d be great :)

**Comment by citizen428** [comment]
@flash-gordon Done. I hope I didn't miss anything...

**Comment by scepticulous** [comment]
Great change. This makes the class names more intuitive and thus onboarding will now be way easier. :)

**Comment by flash-gordon** [comment]
@citizen428 OK, thank you! I merged it to a separate going to test and polish things, update dry-t/dry-v integration, finish my fixed Left PR I opened a few months ago. If didn't forget something I should able to push a new release after that

**Comment by citizen428** [comment]
@flash-gordon Great, thanks! BTW was there anything else I should have done in regards to polishing things, just so I know for the future. 😃

**Comment by flash-gordon** [comment]
@citizen428 I made it work with the existing dry-transaction release, that's basically it

**Pull request merged**